text/0117-typo-setting-api.md

```diff
@@ -0,0 +1,295 @@
+
+### 3.1.2. `disabledAttributes`
+
+- Type: Array of String
+- Required: False
+- Default: `[]`
+
+`disabledAttributes` disable the typo tolerance feature on the specified attributes list.
+
+Any attributes defined in `disabledAttributes` won't have their values matched by the typo tolerance.
+
+#### 3.1.2.1. Example
+
+Given `["title"]` as `disabledAttributes` and the following document
+
+```json
+{
+    "id": 0,
+    "title": "Hey World"
+}
+```
+
+The engine won't try to match query term with typos with values of the `title` attribute at search time to match documents.
+
+- Typing `Warld` won't bring the document as a candidate for the search results.
+
+### 3.1.3. `disabledWords`
+
+- Type: Array of String
+- Required: False
+- Default: `[]`
+
+`disabledWords` disable the typo tolerance feature on a list of search query terms.
+
+> This field is case insensitive since the document attributes values are lowercased and de-unicoded internally.
+
+#### 3.1.3.1. Example
+
+If `Javascript` is specified in `disabledWords`, the engine won't apply the typo tolerance on the query term `Javascript` or `javascript` if typed at search time to match documents.
+
+### 3.1.4. `minWordSizeForTypos`
+
+- Type: Object
+- Required: False
+
+#### 3.1.4.1. `minWordSizeForTypos` Object Definition
+
+| Field                      | Type            | Required |
+|----------------------------|-----------------|----------|
+| [oneTypo](#3142-onetypo)   | Integer         | False    |
+| [twoTypos](#3143-twotypos) | Integer         | False    |
+
+#### 3.1.4.2. `oneTypo`
+
+- Type: Integer
+- Required: False
+- Default: `5`
+
+`oneTypo` customize the minimum size for a word to tolerate 1 typo.
+
+> Given the default value `5`, the search engine starts to apply typo tolerance on a query term if its length is at least equal to 5 characters.
+
+> See [2.1. Typos calculation section](#21-typos-calculation)
+
+##### 3.1.4.2.1 Example
+
+Given `5` as `oneTypo` and the following document
+
+```json
+{
+    "id": 0,
+    "title": "Hey World"
+}
+```
+
+- Typing `World` with 1 typo, e.g. `Warld` will match `World`. It accepts 1 typo since `World` size is made of `5` chars.
+- Typing `Hey` with 1 typo, e.g. `Hoy` won't match `Hey`. It accepts 0 typo since `Hey` size is made of `3` chars.
+
+#### 3.1.4.3. `twoTypos`
+
+- Type: Integer
+- Required: False
+- Default: `9`
+
+`twoTypos` customize the minimum size for a word to tolerate 2 typos.
+
+> Given the default value `9`, the search engine handles up to 2 typos on a query term if its length is at least equal to 9 characters.
+
+> See [2.1. Typos calculation section](#21-typos-calculation)
+
+##### 3.1.4.3.1. Example
+
+Given `3` for `oneTypo` and `5` as `twoTypos` and the following document
+
+```json
+{
+    "id": 0,
+    "title": "Hey World"
+}
+```
+
+- Typing `World` with 2 typos, e.g. `Warrld` will match `World`. It accepts 1 typo up to 2 typos since `World` size is made of `5` chars.
```

Comment:
```suggestion
- Typing `World` with 2 typos, e.g. `Warrld` will match `World`. It accepts up to 2 typos since `World` size is made of `5` chars.
```